app/controllers/concerns/application_shared.rb

```diff
@@ -15,4 +15,35 @@ def set_timezone
     # Reset timezone for the next thread
     Time.zone = default_timezone
   end
-end
\ No newline at end of file
+
+  private
+
+  # Get the allowed parameters from the Apipie documentation in the API controllers.
+  def safe_params
+    apipie_params = Apipie.app.get_method_description("v2##{controller_name}", action_name).to_json[:params]
```

Comment:
This disables updating nested attributes (e.g., ```environments```).
In log we get `Unpermitted parameters: environment_names`, though it works on develop.

We should also add [raise an error on unpermitted params](https://github.com/theforeman/foreman/pull/1819/files#diff-a2f09e1db7a47eb67c5849478a97f3d7R48) for development purposes